server/Models/userModel.ts

```diff
@@ -0,0 +1,56 @@
+const bcrypt = require('bcrypt');
+import { UserType } from '../../types';
+
+const SALT_WORK_FACTOR: number = 10; // Use number type for numeric constant
+
+const myURI: string =
+  'mongodb+srv://codesmithbv:ax6POS3lHiGRGPfO@kalecluster.3wl6slc.mongodb.net/';
+
+import mongoose from 'mongoose';
+const { Schema } = mongoose;
+
+try {
+  mongoose.connect(myURI);
+  console.log('connected to mongodb');
+} catch (error) {
+  console.log(error);
+}
+
+const userSchema = new Schema<UserType>({
+  // Use generic type argument for UserType interface
+  username: { type: String, required: true, unique: true },
```

Comment:
update this to 

email:
name: 
password:

Reply:
updated!